internal/gatewayapi/backendtlspolicy.go

```diff
@@ -49,22 +55,48 @@ func (t *Translator) applyBackendTLSSetting(
 		return nil, err
 	}
 
-	if tlsBundle, err = t.applyEnvoyProxyBackendTLSSetting(upstreamConfig, resources, envoyProxy); err != nil {
+	if tlsBundle, err = t.applyCommonBackendTLSConfig(upstreamConfig, backendRef, backendNamespace, resources); err != nil {
+		return nil, err
+	}
+
+	if tlsBundle, err = t.applyEnvoyProxyBackendTLSSetting(tlsBundle, resources, envoyProxy); err != nil {
 		return nil, err
 	}
 	return tlsBundle, nil
 }
 
-func (t *Translator) processBackendTLSConfig(
+func (t *Translator) applyCommonBackendTLSConfig(
+	tlsConfig *ir.TLSUpstreamConfig,
 	backendRef gwapiv1.BackendObjectReference,
 	backendNamespace string,
 	resources *resource.Resources,
 ) (*ir.TLSUpstreamConfig, error) {
+	if tlsConfig == nil || KindDerefOr(backendRef.Kind, resource.KindService) != resource.KindBackend {
+		return tlsConfig, nil
+	}
+
 	backend := resources.GetBackend(backendNamespace, string(backendRef.Name))
 	if backend == nil {
 		return nil, fmt.Errorf("backend %s not found", backendRef.Name)
 	}
 	if backend.Spec.TLS == nil {
+		return tlsConfig, nil
+	}
+
+	tlsConfig.InsecureSkipVerify = ptr.Deref(backend.Spec.TLS.InsecureSkipVerify, false)
```

Comment:
The TLS setting in the Backend is supposed to be used by the `DynamicResolver` type only.